File: HackerRank/ArraysAndSorting/InsertionSort.java
import java.util.Scanner;


public class InsertionSort {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		int n = sc.nextInt();
		int[] arr = new int[n];

		for(int i = 0; i < n; i++) {
			arr[i] = sc.nextInt();
		}

		int val = arr[n-1];

		int i = n-2;
		while(i >= 0) {
			if(arr[i] > val) {
				arr[i+1] = arr[i];
				print(arr);
			} else {
				arr[i+1] = val;
				print(arr);
				break;
			}
			i--;
		}
		if(i == -1) {
			arr[0] = val;
			print(arr);
		}
	}

	public static void print(int[] arr) {
		for(int i = 0; i < arr.length; i++) {
			System.out.print(arr[i] + " ");
		}
		System.out.println();
	}
}

File: HackerRank/ArraysAndSorting/InsertionSort.cpp
#include <iostream>
#include <vector>
using namespace std;

void print(vector<int> arr) {
	for(int i = 0; i < arr.size(); i++)
		cout << arr[i] << " ";
	cout << endl;
}

void insertionSort(vector<int> arr) {
	int arr_size = arr.size();
	int val = arr[arr_size-1];

	int i = arr_size-2;
	while(arr[i] > val && i >= 0) {
		arr[i+1] = arr[i];
		print(arr);
		i--;
	}

	arr[i+1] = val;
	print(arr);
} 

int main() {
	int n;
	cin >> n;

	vector<int> arr;

	for(int i = 0; i < n; i++) {
		int val;
		cin >> val;
		arr.push_back(val);
	}

	insertionSort(arr);

	return 0;
}